.travis.yml

```diff
@@ -12,6 +12,7 @@ env:
     - TEST=python NOSE_DIVIDED_WE_RUN=bf JS_SETUP=yes
     - TEST=python-sharded-and-javascript JS_SETUP=yes TEST_MAKE_REQUIREMENTS=yes
   global:
+    - REUSE_DB=true  # can't drop the citus DB since it's the postgres database
```

Comment:
Is the use of `postgres` a citus requirement or part of our setup? It seems bad to not have the ability to completely drop and re-create all database state when things go wrong, at least when testing locally. Is the `postgres` database used for application/test data, or is it just extension config stuff in there?

Reply:
It would be great to find a way around this, especially for local testing. This is what Django would need to do:
1. Create the DB on the coordinator (easy)
2. Create the DB on the worker nodes
  - This would mean adding the DB configs to the settings file
3. Create the citus extension in all above DBs
  - Can be done with a DB migration
4. Add the worker nodes to the coordinator
  - Can also be done in a migration

So to do this in Django we'd need to have all the DB configs in the settings and we'd need 2 apps with migrations (one for the worker nodes and one for the coordinator) along with code in the DB router.

I started doing that but it's a lot of extra code etc. that's only necessary for tests. I'm open to ideas though.

Reply:
> I started doing that but it's a lot of extra code etc. that's only necessary for tests.

So it sounds possible, but it's extra work and adds a bunch more code to maintain.

However, the alternative is that some database state will never be reset, even when running tests with `--reusedb=reset`. Also, when running tests locally in addition to a local HQ environment for manual testing, the tests and the local HQ environment will share the same citusdb database. Both of these seem important/bad enough that the extra work and code seems worth it to me.

Since it's for testing only, could the django apps and other code needed for this be added in `./testapps/`?